libs/sqlserver/langchain_sqlserver/vectorstores.py

```diff
@@ -225,31 +229,86 @@ def __init__(
         self._embedding_store = self._get_embedding_store(self.table_name, self.schema)
         self._create_table_if_not_exists()
 
-    def _can_connect_with_entra_id(self) -> bool:
-        """Determine if Entra ID authentication can be used.
+    def _get_connection_url(self, conn_string: str) -> str:
+        if conn_string is None or len(conn_string) == 0:
+            logging.error("Connection string value is None or empty.")
+            raise ValueError("Connection string value cannot be None.")
+
+        args = conn_string.split(";")
+        arg_dict = {}
+        for arg in args:
+            if "=" in arg:
+                # Split into key value pairs by the first positioned `=` found.
+                # Key-Value pairs are inserted into the dictionary.
+                #
+                key, value = arg.split("=", 1)
+                arg_dict[key.lower().strip()] = value.strip()
 
-        Check the components of the connection string to determine
-        if connection via Entra ID authentication is possible or not.
+        try:
+            # This will throw a key error if server or database keyword
+            # is not present in arg_dict from the connection string.
+            #
+            database = arg_dict.pop("database")
 
-        The connection string is of expected to be of the form:
-            "mssql+pyodbc://username:password@servername/dbname?other_params"
-        which gets parsed into -> <scheme>://<netloc>/<path>?<query>
-        """
-        parsed_url = urlparse(self.connection_string)
+            # If `server` is present in the dictionary, we split by
+            # `,` to obtain host and port details.
+            #
+            server = arg_dict.pop("server").split(",", 1)
+            server_host = server[0]
+            server_port = None
 
-        if parsed_url is None:
-            logging.error("Unable to parse connection string.")
-            return False
+            # Check if port is provided in server details,if true,
+            # cast value to int if possible.
+            #
+            if len(server) > 1 and server[1].isdigit():
+                server_port = int(server[1])
 
-        if (parsed_url.username and parsed_url.password) or (
-            "trusted_connection=yes" in parsed_url.query.lower()
+        except KeyError as k:
```

Comment:
The exception should also catch specific errors where the connection string has Database and Server, but the **server_host** is invalid, and incorrectly formatted conn_string?